### PR TITLE
exclude overlapping samples and patients by default for survival and clinical tab

### PR DESCRIPTION
**Pull request by dippindots**
This pr includes these changes:
1. Change option text from  `Exclude overlapping samples` to `Exclude overlapping samples and patients`
2. For survival and clinical tab, exclude overlapping samples and patients by default and show alert `We exclude overlapping samples and patients by default to display this tab`
3. Don't grey out the clinical tab and always show clinical table

Fix # https://github.com/cBioPortal/cbioportal/issues/6010.

### CODE REVIEW

**Comment by kalletlak** [comment]
@dippindots I think even the underlying logic should be updated. I mean excluding overlapping patients

**Comment by dippindots** [comment]
@kalletlak Thanks for your prompt review, I should ask you before do this. I will make the change and tell you when it is ready, thanks!

**Comment by adamabeshouse** [comment]
@kalletlak patients are already excluded independently of samples

**Comment by kalletlak** [comment]
@adamabeshouse yeah true

File: src/pages/groupComparison/GroupComparisonStore.ts [hunk 1]
```diff
@@ -211,6 +211,21 @@ export default class GroupComparisonStore {
         invoke:()=>Promise.resolve(this.availableGroups.result!.filter(group=>this.isGroupActive(group)))
     });
 
+    readonly filteredAvailableGroups = remoteData<OverlapFilteredComparisonGroup[]>({
```
Comment (adamabeshouse) [inline src/pages/groupComparison/GroupComparisonStore.ts]: lets change `filtered` to `nonOverlapping`, it's a bit more specific

Comment (adamabeshouse) [reply src/pages/groupComparison/GroupComparisonStore.ts]: also lets use this in the body of `availableGroups` in the case `OverlapStrategy.EXCLUDE`

**Pull request merged**